internal/driver/gomobile/window.go

```diff
@@ -130,6 +135,15 @@ func (w *window) Hide() {
 }
 
 func (w *window) Close() {
+	if w.willClose != nil {
```

Comment:
I think this will get into a loop here - the willClose check should not be on the Close() call or you will get in a loop of not closing I think?

Reply:
yes, that was the purpose of `closing` in my initial suggestion. I had a hope you do not like this approach because it's not covering programmatic window closing

Reply:
Maybe we should retain the original logic, i.e. `Close()` means unconditionally close the window, and introduce `TryClose()` method that relies on `willClose`?

Reply:
then we have different behavior closing window using window manager and from the code

Reply:
Isn't that how glfw works? User attempting to close the window can be vetoed but the code executing Close() is an instruction to be respected?

>  introduce TryClose() method that relies on willClose

There is no need to expose a new method here, as the gomobile driver draws the window headers so can call any internal method it wants when the close button is tapped...